src/CommunityToolkit.Maui/Extensions/PopupExtensions.shared.cs

```diff
@@ -0,0 +1,122 @@
+﻿using CommunityToolkit.Maui.Core;
+using CommunityToolkit.Maui.Views;
+
+namespace CommunityToolkit.Maui.Extensions;
+
+/// <summary>
+/// Popup extensions.
+/// </summary>
+public static class PopupExtensions
+{
+	/// <summary>
+	/// Shows a popup with the specified options.
+	/// </summary>
+	/// <param name="page">Current page</param>
+	/// <param name="view">Popup content</param>
+	/// <param name="options"><see cref="IPopupOptions"/></param>
+	/// <returns><see cref="PopupResult"/></returns>
+	/// <remarks>This is an <see keyword="async"/> <see keyword="void"/> method. Use <see cref="ShowPopupAsync(Page,View,CommunityToolkit.Maui.IPopupOptions?,CancellationToken)"/> to <see keyword="await"/> this method and return <see cref="PopupResult{T}"/> </remarks>
+	public static void ShowPopup(this Page page, View view, IPopupOptions? options = null)
+	{
+		ArgumentNullException.ThrowIfNull(page);
+
+		ShowPopup(page.Navigation, view, options);
+	}
+
+	/// <summary>
+	/// Shows a popup with the specified options.
+	/// </summary>
+	/// <param name="navigation">Popup parent</param>
+	/// <param name="view">Popup content</param>
+	/// <param name="options"><see cref="IPopupOptions"/></param>
+	/// <returns><see cref="PopupResult"/></returns>
+	/// <remarks>This is an <see keyword="async"/> <see keyword="void"/> method. Use <see cref="ShowPopupAsync(Page,View,CommunityToolkit.Maui.IPopupOptions?,CancellationToken)"/> to <see keyword="await"/> this method</remarks>
+	public static async void ShowPopup(this INavigation navigation, View view, IPopupOptions? options = null)
+	{
+		ArgumentNullException.ThrowIfNull(navigation);
+		ArgumentNullException.ThrowIfNull(view);
+
+		var popupPage = new PopupPage(view, options ?? PopupOptions.Empty);
+
+		await navigation.PushModalAsync(popupPage, false);
```

Comment:
since this is `async void`,  if an `exception` happens during the modal navigation it will break the app, so I would say we should add an try/catch here. I can't see a good reason to crash an app if a modal fails to show up.

Reply:
The Exception will be thrown by the .NET MAUI framework, not us. If an exception is thrown by `PushModalAsync()`, something else in the app is already broken; catching and subsequently hiding the exception from users will further break their app.

Since we don't throw the Exception and we don't control when it is thrown, we should propagate the exception up to the developer.

Reply:
That said, I'm happy to change my mind if anyone can demonstrate a use-case where `PushModalAsync()` throws an Exception that is the fault of `CommunityToolkit.Maui`